tensornetwork/backends/shell/shell_backend.py

```diff
@@ -107,14 +107,22 @@ def rq_decomposition(self, tensor: Tensor,
     r = ShellTensor((center_dim,) + right_dims)
     return q, r
 
-  def concat(self, values: Sequence[Tensor], axis: int) -> Tensor:
+  def shape_concat(self, values: Sequence[Tensor], axis: int) -> Tensor:
     shape = values[0].shape
     if axis < 0:
       axis += len(shape)
     concat_size = sum(v.shape[axis] for v in values)
     new_shape = shape[:axis] + (concat_size,) + shape[axis + 1:]
     return ShellTensor(new_shape)
 
+  def concat(self, values: Sequence[Tensor], axis: int = 0) -> Tensor:
+    new_shape = None
+    if axis == 0:
+      new_shape = ShellTensor(values)
+    else:
+      new_shape = self.shape_concat(values, axis)
+    return new_shape
+

```

Comment:
I don't this this is correct. `ShellTensor` is a tensor type that only stores its shape and has no concrete values, so when you do a normal concatenation of multiple `ShellTensors`, you'll want to add the value of that specified axis together.

Though this function really isn't important at all. Just throw a `NotImplementedError` and we'll add it when we actually need it.